src/solver.rs

```diff
@@ -647,18 +647,24 @@ impl Solver {
 
     /// Compute the entering variable for a pivot operation.
     ///
-    /// This method will return first symbol in the objective function which
+    /// This method will return a symbol in the objective function which
     /// is non-dummy and has a coefficient less than zero. If no symbol meets
     /// the criteria, it means the objective function is at a minimum, and an
     /// invalid symbol is returned.
     /// Could return an External symbol
     fn get_entering_symbol(objective: &Row) -> Symbol {
+        let mut entering = Symbol::invalid();
+        let mut min_id = usize::MAX;
+
         for (symbol, value) in &objective.cells {
             if symbol.kind() != SymbolKind::Dummy && *value < 0.0 {
-                return *symbol;
+                if symbol.id() < min_id {
+                    min_id = symbol.id();
+                    entering = *symbol;
+                }
             }
         }
-        Symbol::invalid()
+        entering
```

Comment:
Can this use [min_by_key](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min_by_key). E.g. something like:

```suggestion
objective.cells
    .iter()
    .filter(|(symbol, value)| symbol.kind() != SymbolKind::Dummy && *value < 0.0)
    .min_by_key(|(symbol, _)| symbol.id())
    .unwrap_or(Symbol::invalid())
```

Reply:
Probably yes, this is a dirty branch, I'm just exploring what works at this moment so I want to keep the changes minimal.